File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        map<int,int> mp;
        for(int i = 0; i < nums.size(); i++){
            mp[nums[i]]++;
        }
        int count = 1;
        int res = 0;
        for(auto n : mp){
            if(mp.find(n.first+1)!=mp.end()){
                count++;
                continue;
            }
            
            res = max(count, res);
            count = 1;
        }
        return res;
    }
};

File: Data Structures & Algorithms/maximum-ascending-subarray-sum/submission-1.py
class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        res = msum = nums[0]
        for i in range(1, len(nums)):
            if nums[i] > nums[i-1]:
                msum += nums[i]
                res = max(res, msum)
            else:
                msum = nums[i]
        return res